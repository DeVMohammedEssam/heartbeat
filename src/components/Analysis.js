import React from 'react';
import Navbar from './layout/Navbar';
import axios from "axios"
import {url} from "../config"
class Analysis extends React.Component{
    state={
      postive:0
      ,negative:0,
      all:0
    }
       componentDidMount(){
        axios.get(url+"analysis").then((response)=>{
                this.setState({
                    ...response.data
                })
        })
    }
  render(){
    return (
        <React.Fragment>
            <Navbar />
            <section className="analysis">
                <h3 className="analysis__title">Analysis</h3>
                <div className="analysis__cards-container">
                    <div className="analysis__card">
                        <div className="analysis__card__icon-container">
                            <img src="/images/users-group.png" alt="" />
                        </div>
                        <div className="analysis__card__number">{this.state.all}</div>
                    </div>
                    <div className="analysis__card">
                        <div className="analysis__card__icon-container">
                            <img src="/images/patient.png" alt="" />
                        </div>
                        <div className="analysis__card__number">{this.state.postive}</div>
                    </div>
                    <div className="analysis__card">
                        <div className="analysis__card__icon-container">
                            <img src="/images/verified.png" alt="" />
                        </div>
                        <div className="analysis__card__number">{this.state.negative}</div>
                    </div>
                </div>
            </section>
        </React.Fragment>
    );
  }
}

export default Analysis;