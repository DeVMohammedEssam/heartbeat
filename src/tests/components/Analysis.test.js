import { shallow } from "enzyme";
import React from "react";
import Analysis from "../../components/Analysis.js"
test("should render analysis page correctly", () => {
    const wrapper = shallow(<Analysis />);
    expect(wrapper).toMatchSnapshot();

})