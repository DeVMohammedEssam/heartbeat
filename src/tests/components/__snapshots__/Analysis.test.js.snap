// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render analysis page correctly 1`] = `
<Fragment>
  <ConnectFunction />
  <section
    className="analysis"
  >
    <h3
      className="analysis__title"
    >
      Analysis
    </h3>
    <div
      className="analysis__cards-container"
    >
      <div
        className="analysis__card"
      >
        <div
          className="analysis__card__icon-container"
        >
          <img
            alt=""
            src="/images/users-group.png"
          />
        </div>
        <div
          className="analysis__card__number"
        >
          25,22l
        </div>
      </div>
      <div
        className="analysis__card"
      >
        <div
          className="analysis__card__icon-container"
        >
          <img
            alt=""
            src="/images/patient.png"
          />
        </div>
        <div
          className="analysis__card__number"
        >
          25,22l
        </div>
      </div>
      <div
        className="analysis__card"
      >
        <div
          className="analysis__card__icon-container"
        >
          <img
            alt=""
            src="/images/verified.png"
          />
        </div>
        <div
          className="analysis__card__number"
        >
          25,22l
        </div>
      </div>
    </div>
  </section>
</Fragment>
`;
